pkgs/by-name/dm/dm-media-converter/package.nix

```
@@ -0,0 +1,57 @@
+{
+  autoPatchelfHook,
+  fetchurl,
+  ffmpeg,
+  lib,
+  libqt5pas,
+  libX11,
+  numactl,
+  qtbase,
+  stdenv,
+  wrapQtAppsHook,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dm-media-converter";
+  version = "2.5.5";
+
+  src = fetchurl {
+    url = "https://drive.usercontent.google.com/download?id=1xBg3mCfvl8TlB4CyacD8YQN0CvHcdn7I&export=download&confirm=t";
+    sha256 = "sha256-Jn/Zf5fKIzKjP7GoMuOFq4QlG29azcoEAT/qySAofLA=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    libqt5pas
+    libX11
+    ffmpeg
+    qtbase
+    numactl
+  ];
+
+  sourceRoot = ".";
+
+  unpackCmd = "tar -xzf $src --strip-components=1";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+
+    mv dmMediaConverter bin --target-directory $out/bin
```

```suggestion
    mv dmMediaConverter --target-directory $out/bin

    mkdir "$out"/bin/x86_64-linux
    ln -s "${lib.getExe ffmpeg}" "$out"/bin/x86_64-linux/ffmpeg
    ln -s "${lib.getExe ffmpeg}" "$out"/bin/x86_64-linux/ffmpeg.mfx
	ln -s "${lib.getExe ffprobe}" "$out"/bin/x86_64-linux/ffprobe
```
Instead of using the ancient ffmpeg 4.2.1 included in the app I would suggest to use our ffmpeg.

---

i can't get this one to build. Even after adding `ffprobe` to the dependencies

---

You would need to add ffprobe at the top too

---

Or which error are you getting?

---

At first I get `undefined variable 'ffprobe'` for line 50. After adding it to the file dependencies at the top, all I get is this cryptic error `lib.customisation.callPackageWith: Function called without required argument "ffprobe"` for the line I added the `ffprobe` dependency to. along side a presumably unhelpful "did you mean `lprobe`" message

---

Oh right. I forgot, ffprobe is also supplied by ffmpeg.

So you need `${lib.getExe' ffmpeg "ffprobe"}`.

---

now i get this "missing required binaries" message from dmMediaConverter. After trying to add a job, it says it cant find "ffprobe"

this appears to be related to the bin directory included in the portable being relative the the executable of dmMediaConverter.

I now propose this for the install phase and am ready to commit:
```nix
    runHook preInstall

    mkdir -p "$out"/bin/bin/x86_64-linux

    mv dmMediaConverter "$out"/bin

    ln -s "${lib.getExe ffmpeg}" "$out"/bin/bin/x86_64-linux/ffmpeg
    ln -s "${lib.getExe ffmpeg}" "$out"/bin/bin/x86_64-linux/ffmpeg.mfx
    ln -s "${lib.getExe' ffmpeg "ffprobe"}" "$out"/bin/bin/x86_64-linux/ffprobe

    runHook postInstall
```

after making these changes, I have gotten the package to build, run, and re-encode a test file

---

Looks good to me